lib/matplotlib/tests/test_image.py

```diff
@@ -993,6 +993,13 @@ def test_empty_imshow(make_norm):
         im.make_image(fig._cachedRenderer)
 
 
+def test_imshow_float16():
+    fig, ax = plt.subplots()
+    ax.imshow(np.zeros((3, 3), dtype=np.float16))
```

Comment:
This warns too?  Should you check for the warning?

Reply:
No, this doesn't warn because it's an upcast (just like we don't warn for int->float).
(Unchecked warnings normally cause tests to fail, so this also tests that no warning is emitted.)